crates/wasmparser/src/validator/core.rs

```diff
@@ -569,6 +569,23 @@ impl Module {
                 offset,
             )?;
         }
+        if !(features.function_references()
+            || features.gc()
+            || features.component_model()
+            || features.stack_switching())
```

Comment:
One improvement that I just see here is that instead of disallow-listing certain proposals, we could allow-list all the proposals of which we know this fast validation is applied correctly. I think this is a more future proof solution. We could construct a `const FAST_VALIDATE_FEATURES: WasmFeatures = ...` and then just check via a single
```rust
FAST_VALIDATE_FEATURES.contains(features)
```

Reply:
I just implemented this and am convinced that this is a much better approach for maintainability and future proofing of continuous feature development.